hail/python/test/hailtop/test_aiogoogle.py

```diff
@@ -283,24 +319,25 @@ async def test_multi_part_create(local_filesystem, permutation):
     for b in part_data:
         part_start.append(s)
         s += len(b)
-    print(part_start)
 
     path = f'{base}a'
-    async with await fs.multi_part_create(path, len(part_data)) as c:
-        async def create_part(i):
-            async with await c.create_part(i, part_start[i]) as f:
-                await f.write(part_data[i])
-
-        if permutation:
-            # do it in a fixed order
-            for i in permutation:
-                await create_part(i)
-        else:
-            # do in parallel
-            await asyncio.gather(*[
-                create_part(i) for i in range(len(part_data))])
-
-    expected = b''.join(part_data)
-    async with await fs.open(path) as f:
-        actual = await f.read()
-    assert expected == actual
+    sema = asyncio.Semaphore(50)
+    async with sema:
+        async with await fs.multi_part_create(sema, path, len(part_data)) as c:
+            async def create_part(i):
+                async with await c.create_part(i, part_start[i]) as f:
+                    await f.write(part_data[i])
```

Comment:
It's not quite clear to me that these tests cover the case where we have > 32 partitions to make sure the GoogleStorageMultiPartCreate is working correctly with the tree compose and the removal of files.

Reply:
Agreed, added a new test: test_multi_part_create_many